plugins/elanfp/fu-elanfp-file-control.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2021
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "fu-elanfp-file-control.h"
+
+#define SIZE_IDENTIFY_PACKET 16
+
+#define S2F_TAG_FIRMWAREVERSION 0x00
+#define S2F_TAG_CFU_OFFER_A	0x72
+#define S2F_TAG_CFU_OFFER_B	0x73
```

Comment:
So this looks a lot like CFU -- but is this standard Microsoft CFU or something Elan-specific? Some details would be great.

Reply:
This is standard Microsoft CFU.

Reply:
So I'm wondering if we should call the plugin "cfu" rather than "elanfp" -- it seems like other devices would want to use this in the future.

Reply:
Can always rename it later if there is a second device and the code really is identical

Reply:
The reason I say this is that on Windows there is a concept of using an "Extension INF" and the specifics of how that ends up reflecting on fwupd might not be 1:1 mapping.

Reply:
Sorry , let me clarify it. We are not use standard Microsoft CFU. The difference is that CFU is using HID communication, but our is for USB device. You can treat as Elan-specific.